### DISCUSSION TITLE
How deploy app on out own VDS without development mode?

### DISCUSSION THREAD

**Discussion by unameisfine** [AllYourBot/hostedgpt#492] [General]
How to deploy the application on our own server in a production environment? Otherwise, there is too much debugging and logging. Currently, the docker-compose specifies "target: development".

**Comment by keithschacht** [comment]
Ah, I haven’t yet dug into creating a deploy process for any VDS. This issue https://github.com/AllYourBot/hostedgpt/issues/161 is where I was tracking it and one other person was going to work on that, using Kamal, since it provides a nice set of rails for it. Maybe chime in on that issue with any specific considerations that are important to you. It’s good for us to know we’re covering our basis.

If logging is your only concern, you could just add RAILS_ENV=production as another env file. However, I suspect there are other tweaks that are needed besides just that. While we continue to work through deploying to any VDS, I do have it set up to deploy easily to Render, Fly, and Heroku. Those are al pretty well tested.

**Comment by ceicke** [comment]
I always consider docker-compose to be only there for development. I have this project run on my VDS as a Rails standalone app. You can go all-in with apache and passenger, but I am simply having a tmux session running with:

`RAILS_ENV=production RUN_SOLID_QUEUE_IN_PUMA=true rails s -p 8081`

And I use my Nginx simply as a proxy to it:

```
<VirtualHost *:443>
        ServerName chat.${maindomain}
        ServerAlias chat.${secondarydomain}
        ProxyPreserveHost On
        ProxyPass / http://localhost:8081/
        ProxyPassReverse / http://localhost:8081/
        RequestHeader set X-Forwarded-Proto "https"
        <Location /cable>
                ProxyPreserveHost On
                ProxyPass ws://localhost:8081/cable
                ProxyPassReverse ws://localhost:8081/cable
        </Location>

        Include /etc/letsencrypt/options-ssl-apache.conf
        SSLCertificateFile /etc/letsencrypt/live/chat.${maindomain}/fullchain.pem
        SSLCertificateKeyFile /etc/letsencrypt/live/chat.${maindomain}/privkey.pem
</VirtualHost>
```

Hope it helps.

**Reply by keithschacht** [reply]
@ceicke thank you for this. I think this could help out a bunch of people so I've updated the README to explain the basics and I've included this nginx example config.
